src/applications/personalization/profile360/tests/util/index.unit.spec.js

```diff
@@ -103,4 +107,51 @@ describe('profile utils', () => {
       expect(eventDataObject).to.deep.equal(badHomePhoneDataObject);
     });
   });
+
+  describe('PaymentInformation error parsing methods', () => {
+    it('hasFlaggedForFraudError returns true on error', () => {
+      const errors = [
+        {
+          title: 'Account Flagged',
+          detail: 'The account has been flagged',
+          code: '136',
+          source: 'EVSS::PPIU::Service',
+          status: '422',
+          meta: {
+            messages: [
+              {
+                key: 'cnp.payment.flashes.on.record.message',
+                severity: 'ERROR',
+                text: 'Flashes on record',
+              },
+            ],
+          },
+        },
+      ];
+      expect(hasFlaggedForFraudError(errors)).to.equal(true);
+    });
+
+    it('hasInvalidHomePhoneNumberError returns false if text does not contain night phone', () => {
```

Comment:
This tests the possible false positive condition Erik described in the ticket.  We want to test the key and the text in the same pass instead of individually.

Reply:
So the case I'm worried about, which admittedly not likely to happen because I don't know if we will ever get back multiple entries in that `meta.messages` array, is if the `meta.messages` array looked like this:
```
[
  {
    key: 'cnp.payment.generic.error.message',
    severity: 'ERROR',
    text: "some error we don't know about",
  },
  {
    key: 'unknown.key',
    severity: 'ERROR',
    text:
      'Generic CnP payment update error. Update response: Update Failed: Night area number is invalid, must be 3 digits',
  },
]
```
In this case the `hasInvalidHomePhoneNumberError()` should return `false` because although the `cnp.payment.generic.error.message` code appears on a message object and the `night area number` appears in a message object's `text`, they appear on _different_ message objects.

This test you wrote is totally valid but it doesn't test for this hypothetical false-positive I noticed we could get. Again, I'm wanting this to be coded overly defensively, but might as well do it right, right? :)

Basically the parsing logic needs to be smarter and check that both the `key` and `text` of _the same message object_ match what we are looking when checking which error message we should show.

Reply:
@erikphansen actually, it does!  https://github.com/department-of-veterans-affairs/vets-website/pull/10951/files#diff-252aebd3806ab343cc3a5fc208865419R27 . I changed the method thats evaluating the errors and if the message text key/value is present, it evaluates it at the same time as the error key.

Reply:
@erikphansen I added a test case to cover this ^^